pkg/accesscode/accesscode.go

```diff
@@ -34,36 +36,28 @@ func (acc AccessCodeClient) GetAccessCodes(codes []string) ([]hfv1.AccessCode, e
 		return nil, fmt.Errorf("code list passed in was less than 0")
 	}
 
-	accessCodeList, err := acc.hfClientSet.HobbyfarmV1().AccessCodes(util.GetReleaseNamespace()).List(acc.ctx, metav1.ListOptions{})
+	acReq, err := labels.NewRequirement(util.AccessCodeLabel, selection.In, codes)
```

Comment:
most important change.
Query AccessCodes based on label instead of iterating all accesscodes.